components/results_table/commons/results_table.lua

```diff
@@ -101,8 +109,203 @@ function BaseResultsTable:getQueryType()
 	error('Invalid querytype "' .. (args.querytype or '') .. '"')
 end
 
+function BaseResultsTable:create()
+	local data = self.args.data or self:queryData()
+
+	if Table.isEmpty(data) then
+		return
+	end
+
+	table.sort(data, function(placement1, placement2) return placement1.date > placement2.date end)
+
+	if self.config.onlyAchievements then
+		self.data = {data}
+		return self
+	end
+
+	-- split data into years for non achievements
+	local splitData = {}
+	for _, placementData in ipairs(data) do
+		local year = placementData.date:sub(1,4)
+		if not splitData[year] then
+			splitData[year] = {header = year}
+		end
+
+		table.insert(splitData[year], placementData)
+	end
+
+	self.data = {}
+	for _, dataSet in Table.iter.spairs(splitData, function(tbl, key1, key2) return key1 > key2 end) do
+		table.insert(self.data, dataSet)
+	end
+
+	return self
+end
+
+function BaseResultsTable:queryData()
+	local data = mw.ext.LiquipediaDB.lpdb('placement', {
+		limit = self.config.limit,
+		order = self.config.sort .. ' ' .. self.config.order,
+		conditions = self:buildConditions(),
+	})
+
+	if type(data) ~= 'table' then
+		error(data)
+	end
+
+	return data
+end
+
+function BaseResultsTable:buildConditions()
+	if self.args.customConditions then
+		return self.args.customConditions
+	end
+
+	local conditions = self:buildBaseConditions()
+
+	if self.args.additionalConditions then
+		return conditions .. self.args.additionalConditions
+	end
+
+	return conditions
+end
+
+function BaseResultsTable:buildBaseConditions()
+	local args = self.args
+
+	local conditions = ConditionTree(BooleanOperator.all)
+		:add{self:buildOpponentConditions()}
+
+	if args.game then
+		conditions:add{ConditionNode(ColumnName('game'), Comparator.eq, args.game)}
+	end
+
+	local startDate = args.startdate or args.sdate
+	if startDate then
+		-- intentional > here to keep it as is in current modules
+		-- possibly change to >= later
+		conditions:add{ConditionNode(ColumnName('date'), Comparator.gt, startDate)}
+	end
+
+	local endDate = args.enddate or args.edate
+	if endDate then
+		-- intentional < here to keep it as is in current modules
+		-- possibly change to <= later
+		conditions:add{ConditionNode(ColumnName('date'), Comparator.lt, endDate)}
+	end
+
+	if args.placement then
+		conditions:add{ConditionNode(ColumnName('placement'), Comparator.eq, args.placement)}
+	elseif not Logic.readBool(args.ignorePlacement) then
+		conditions:add{ConditionNode(ColumnName('placement'), Comparator.neq, '')}
```

Comment:
What's this?

Reply:
`[[placement::!]]`, i.e. placement is set
which atm would make it unusable for awards stuff, but is intended for non award stuff

Reply:
will add a todo for now so we look into it once we decided how to handle awards stuff storage wise